imports/api/logic/grad-checker/faculty/computing/computer-science/AY-1415/breadth-depth/focus-area/focusArea-test.js

```diff
@@ -0,0 +1,99 @@
+    dePopulateModuleFixture();
+    dePopulatePlannerFixture(plannerIDs);
+    dePopulateComputerGraphicsFocusAreaRequirementsFixture(focusAreaReqIDs);
+    dePopulateGraduationRequirementsFixture(gradFocusAreaID);
+    done();
+  });
+
+  it ('checks if find focus area MCs return true', function() {
+    const academicCohort = 'AY 2016/2017';
+    const requirementName = 'Computer Science Focus Area' ;
+    const allSemesters = getAllSemestersInPlanner(plannerIDs[0]);
+
+    const focusAreaPrimaryModules = getFocusAreaPrimaryRequirement(focusAreaReqIDs);
+    const focusArea4KModules = getFocusArea4KRequirement(focusAreaReqIDs);
+    const focusAreaNonPrimaryModules = getFocusAreaNonPrimaryRequirement(focusAreaReqIDs);
+    const focusAreaPrimary4KModules = getFocusAreaPrimary4KRequirement(focusAreaReqIDs);
+
+    //const studentFocusAreas = getFocusAreaRequirement(focusAreaReqIDs);
+    const requiredMCs = getGradRequirementMCs(gradFocusAreaID)[requirementName];
+
+    const studentFocusAreas = {
+      focusAreaPrimaryModules: focusAreaPrimaryModules,
+      focusArea4KModules: focusArea4KModules,
+      focusAreaPrimary4KModules: focusAreaPrimary4KModules,
+      focusAreaNonPrimaryModules: focusAreaNonPrimaryModules
+    }
+
+    const focusAreaMCSFulfilment = checkFocusAreaFulfilmentMCs(allSemesters, studentFocusAreas, requiredMCs);
+    assert.isTrue(focusAreaMCSFulfilment.isFulfilled, 'focus area is fulfiled');
+    assert.equal(focusAreaMCSFulfilment.requiredMCs, 24);
+  })
+
+  it ('checks if focus area requirement modules return true', function() {
+    const academicCohort = 'AY 2016/2017';
+    const focusAreaName = 'Computer Graphics and Games';
+    const allSemesters = getAllSemestersInPlanner(plannerIDs[0]);
+    //const focusAreaModules = getFocusAreaRequirement(focusAreaReqIDs);
+
+    //const focusAreaPrimaryModules = focusAreaModules.focusAreaPrimaryModules;
+    //const focusArea4KModules = focusAreaModules.focusArea4KModules;
+    //const focusAreaNonPrimaryModules = focusAreaModules.focusAreaNonPrimaryModules;
+
+    const focusAreaPrimaryModules = getFocusAreaPrimaryRequirement(focusAreaReqIDs);
+    const focusAreaPrimary4KModules = getFocusAreaPrimary4KRequirement(focusAreaReqIDs);
+    const focusArea4KModules = getFocusArea4KRequirement(focusAreaReqIDs);
+    const focusAreaNonPrimaryModules = getFocusAreaNonPrimaryRequirement(focusAreaReqIDs);
+
+    const studentFocusAreas = {
+      focusAreaPrimaryModules: focusAreaPrimaryModules[focusAreaName],
+      focusAreaPrimary4KModules: focusAreaPrimary4KModules[focusAreaName],
+      focusArea4KModules: focusArea4KModules[focusAreaName],
+      focusAreaNonPrimaryModules: focusAreaNonPrimaryModules[focusAreaName]
+    }
+    const markedFocusAreaModulesAndMCs = findFocusAreaModules(focusAreaName, academicCohort, allSemesters, studentFocusAreas, {}, {});
+
+    assert.isTrue(markedFocusAreaModulesAndMCs.isPrimaryTrue, 'primary focus area is fulfiled');
```

Comment:
Becareful of hardcoded test assert values here because it might be hard to change the code next time if requirements change